nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/ListenSlack.java

```diff
@@ -0,0 +1,199 @@
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessSessionFactory;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processors.slack.controllers.SlackConnectionService;
+
+@Tags({"slack", "listen", "json", "events"})
+@CapabilityDescription("Listens on the provided slack connection controller service and " +
+  "forwards the messages in JSON text format.")
+@SeeAlso({SlackConnectionService.class, FetchSlack.class})
+@InputRequirement(Requirement.INPUT_FORBIDDEN)
+public class ListenSlack extends AbstractSessionFactoryProcessor {
+
+  private volatile ProcessSessionFactory processSessionFactory;
+
+  private volatile SlackConnectionService slackConnectionService;
+
+  private volatile List<String> matchingTypes;
+
+  private volatile boolean matchAny;
+
+  static final PropertyDescriptor SLACK_CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+    .name("slack-connection-service")
+    .displayName("Slack Connection ControllerService")
+    .description("A ControllerService that provides a connection to slack service.")
+    .required(true)
+    .identifiesControllerService(SlackConnectionService.class)
+    .build();
+
+  static final PropertyDescriptor MESSAGE_TYPES = new PropertyDescriptor
+    .Builder()
+    .name("message-types")
+    .displayName("Message types")
+    .description("Message types to listen to. It will filter messages with the given types or " +
+      "listen to every type if empty. It is a coma separated list like: message,file_shared")
+    .required(false)
+    .addValidator(Validator.VALID)
+    .build();
+
+  static final Relationship MATCHED_MESSAGES_RELATIONSHIP = new Relationship.Builder()
+    .name("matched")
+    .description("Incoming messages that matches the given types")
+    .build();
+
+  static final Relationship UNMATCHED_MESSAGES_RELATIONSHIP = new Relationship.Builder()
+    .name("unmatched")
+    .description("Incoming messages that does not match the given types")
+    .autoTerminateDefault(true)
+    .build();
+
+  private static final String JSON_OBJECT_TYPE_KEY = "type";
+
+  private static final List<PropertyDescriptor> DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+    SLACK_CONNECTION_SERVICE, MESSAGE_TYPES));
+
+  private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+    MATCHED_MESSAGES_RELATIONSHIP, UNMATCHED_MESSAGES_RELATIONSHIP)));
+
+  @Override
+  public Set<Relationship> getRelationships() {
+    return RELATIONSHIPS;
+  }
+
+  @Override
+  public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+    return DESCRIPTORS;
+  }
+
+  @OnScheduled
+  public void onScheduled(final ProcessContext context) {
+    String messageTypes = context.getProperty(MESSAGE_TYPES).getValue();
```

Comment:
Please don't forget to call `evaluateAttributeExpressions` after we support EL with this property.